tests/sentry/runner/test_initializer.py

```diff
@@ -0,0 +1,173 @@
+    settings.SERVER_EMAIL = 'my-mail-from'
+    settings.EMAIL_SUBJECT_PREFIX = 'my-mail-subject-prefix'
+    settings.FOO_BAR = 'lol'
+
+    bootstrap_options(settings)
+    assert settings.SENTRY_OPTIONS == {
+        'system.secret-key': 'my-system-secret-key',
+        'mail.backend': 'my-mail-backend',
+        'mail.host': 'my-mail-host',
+        'mail.port': 123,
+        'mail.username': 'my-mail-username',
+        'mail.password': 'my-mail-password',
+        'mail.use-tls': True,
+        'mail.from': 'my-mail-from',
+        'mail.subject-prefix': 'my-mail-subject-prefix',
+    }
+
+
+def test_bootstrap_options_no_config_only_sentry_options(settings):
+    "SENTRY_OPTIONS is only declared, but should be promoted into settings"
+    settings.SENTRY_OPTIONS = {
+        'system.secret-key': 'my-system-secret-key',
+        'mail.backend': 'my-mail-backend',
+        'mail.host': 'my-mail-host',
+        'mail.port': 123,
+        'mail.username': 'my-mail-username',
+        'mail.password': 'my-mail-password',
+        'mail.use-tls': True,
+        'mail.from': 'my-mail-from',
+        'mail.subject-prefix': 'my-mail-subject-prefix',
+    }
+
+    bootstrap_options(settings)
+    assert settings.SECRET_KEY == 'my-system-secret-key'
+    assert settings.EMAIL_BACKEND == 'my-mail-backend'
+    assert settings.EMAIL_HOST == 'my-mail-host'
+    assert settings.EMAIL_PORT == 123
+    assert settings.EMAIL_HOST_USER == 'my-mail-username'
+    assert settings.EMAIL_HOST_PASSWORD == 'my-mail-password'
+    assert settings.EMAIL_USE_TLS is True
+    assert settings.SERVER_EMAIL == 'my-mail-from'
+    assert settings.EMAIL_SUBJECT_PREFIX == 'my-mail-subject-prefix'
+
+
+def test_bootstrap_options_missing_file(settings):
+    bootstrap_options(settings, 'this-file-does-not-exist-xxxxxxxxxxxxxx.yml')
+    assert settings.SENTRY_OPTIONS == {}
+
+
+def test_bootstrap_options_empty_file(settings, config_yml):
+    config_yml.write('')
+    bootstrap_options(settings, str(config_yml))
+    assert settings.SENTRY_OPTIONS == {}
+
+
+def test_apply_legacy_settings(settings):
+    settings.SENTRY_USE_QUEUE = True
+    settings.SENTRY_ALLOW_REGISTRATION = True
+    settings.SENTRY_ADMIN_EMAIL = 'admin-email'
+    settings.SENTRY_URL_PREFIX = 'http://url-prefix'
+    settings.SENTRY_SYSTEM_MAX_EVENTS_PER_MINUTE = 10
+    settings.SENTRY_REDIS_OPTIONS = {'foo': 'bar'}
+    settings.SENTRY_OPTIONS = {
+        'mail.from': 'mail-from',
+    }
+    apply_legacy_settings(settings)
+    assert settings.CELERY_ALWAYS_EAGER is False
+    assert settings.SENTRY_FEATURES['auth:register'] is True
+    assert settings.SENTRY_OPTIONS == {
+        'system.admin-email': 'admin-email',
+        'system.url-prefix': 'http://url-prefix',
+        'system.rate-limit': 10,
+        'redis.clusters': {'default': {'foo': 'bar'}},
+        'mail.from': 'mail-from',
+    }
+    assert settings.DEFAULT_FROM_EMAIL == 'mail-from'
+
+
+def test_initialize_app(settings):
```

Comment:
This isn't the most useful test, but during dev, these steps were raising exceptions, so figured it was useful to keep as a sanity check. I plan to expand in this area, it's just a lot of state cases and didn't feel like sinking more time in yet.